cohort-engine/src/main/java/com/ibm/cohort/engine/measure/R4DataProviderFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.measure;
+
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import com.ibm.cohort.engine.measure.cache.CachingRetrieveProvider;
+import com.ibm.cohort.engine.measure.cache.RetrieveCacheContext;
+import org.opencds.cqf.cql.engine.data.CompositeDataProvider;
+import org.opencds.cqf.cql.engine.data.DataProvider;
+import org.opencds.cqf.cql.engine.fhir.model.R4FhirModelResolver;
+import org.opencds.cqf.cql.engine.fhir.retrieve.RestFhirRetrieveProvider;
+import org.opencds.cqf.cql.engine.fhir.searchparam.SearchParameterResolver;
+import org.opencds.cqf.cql.engine.model.ModelResolver;
+import org.opencds.cqf.cql.engine.retrieve.RetrieveProvider;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * <p>An internal class intended to ease the burden of creating the URI to {@link DataProvider} mapping
+ * required by the {@link MeasureEvaluator}.
+ *
+ * <p>This should only be used if the {@link R4MeasureEvaluatorBuilder} is insufficient for your usecase.
+ */
+public class R4DataProviderFactory {
+
+	protected static final String FHIR_R4_URL = "http://hl7.org/fhir";
```

Comment:
```suggestion
	protected static final String FHIR_R4_URL = "http://hl7.org/fhir/R4";
```
?

not sure that it would matter in the end though if you always access it via the _constant_

Reply:
It's important that the URL matches what the cqf-ruler code resolves the `using FHIR` statement to.
https://github.com/DBCG/cqf-ruler/blob/20f09e75ce524d971042e5369afcd138a9dfd079/common/src/main/java/org/opencds/cqf/common/helpers/UsingHelper.java#L19
It's either doing an assumed hardcoding match like this, or allowing ALL using statements to resolve to the provided DataProvider (regardless of whether it's actually using FHIR or not).
I guess I could also go back towards having the DataProvider be created during evaluation, but I would still need to magically know to create an R4 DataProvider when I see that exact URL.